src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newObj = {};
+  const result = [];
+
+  Object.assign(newObj, state);
+
+  for (let i = 0; i < actions.length; i++) {
+    const type = actions[i].type;
+
+    if (type === 'addProperties') {
+      const extraData = actions[i].extraData;
+
+      for (const key in extraData) {
+        newObj[key] = extraData[key];
+      }
+      result.push(Object.assign({}, newObj));
```

Comment:
Just push one time at the end of the iteration after if block

Reply:
if I make an output only once, the last two tests will not pass because they expect an array of objects